optimade/server/config.ini

```diff
@@ -7,8 +7,10 @@ STRUCTURES_COLLECTION = structures
 
 [IMPLEMENTATION]
 PAGE_LIMIT = 500
-VERSION = 0.10.0
+API_VERSION = 0.10.0
```

Comment:
Should this really be a configuration?
Shouldn't this be part of the python code?
E.g. in the top-level __init__ or in the models folder?

Reply:
Just to clarify: I don't think you can really "configure" the API_VERSION in the sense that you can simply change it to run one instance of a server with one version and then another instance with another version.

Perhaps that also solves the problem above.

Reply:
This is the OPTiMaDe version that this implementation/server is adhering to. It is used when initializing the server to set the version (OpenAPI).
Furthermore, it is used as reference for values under the `meta` field, specifically the `api_version` field, and then it's used internally to check against the OPTiMaDe shield/badge I made.

It can be updated/set (also here) using `inv set-optimade-ver --ver="Major.Minor.Patch"`.

We could probably make it a more integral part - but in the end, we should be able to base multiple versions of OPTiMaDe servers on this package. But maybe that assumption has gone out the window?

Reply:
Ok, I understand that such a configuration could in principle make sense, if we had multiple implemented versions of the spec implemented in the same codebase, and the configuration would be choosing one of them.
In that case, however, one should actually make it more complex, since you would need to choose the one for the top-level base URL + which versions to serve using version-specific base URLs.

Currently there is only one, so to me the most sensible thing is that the server simply "reports" the version of the spec implemented in the code.

Consider that configuration files are typically specific to the instance, so if people update the code (which actually updates the implemented spec) and they keep their old configuration file, they will report the wrong API version.

Reply:
Hmm, yeah, perhaps this is a project for the distant future - if it becomes necessary at all.

You make a lot of sense here. I'll try and fix it - make it something non-overridable (in terms of your last example).